sopel/cli/run.py

```diff
@@ -0,0 +1,619 @@
+    except ConfigurationError as e:
+        stderr(e)
+        return ERR_CODE_NO_RESTART
+
+    if config_module.core.not_configured:
+        stderr('Bot is not configured, can\'t start')
+        return ERR_CODE_NO_RESTART
+
+    # Step Three: Manage logfile, stdout and stderr
+    utils.redirect_outputs(config_module, opts.quiet)
+
+    # Step Four: Handle process-lifecycle options and manage the PID file
+    pid_dir = config_module.core.pid_dir
+    pid_file_path = get_pid_filename(opts, pid_dir)
+    old_pid = get_running_pid(pid_file_path)
+
+    if old_pid is not None and tools.check_pid(old_pid):
+        if not opts.quit and not opts.kill and not opts.restart:
+            stderr('There\'s already a Sopel instance running with this config file')
+            stderr('Try using either the `sopel stop` command or the `--restart` option')
+            return ERR_CODE
+        elif opts.kill:
+            tools.stderr(
+                'option -k/--kill is deprecated, '
+                'use `sopel stop --kill` instead')
+            stderr('Killing the Sopel')
+            os.kill(old_pid, signal.SIGKILL)
+            return
+        elif opts.quit:
+            tools.stderr(
+                'options -q/--quit is deprecated, '
+                'use `sopel stop` instead')
+            stderr('Signaling Sopel to stop gracefully')
+            if hasattr(signal, 'SIGUSR1'):
+                os.kill(old_pid, signal.SIGUSR1)
+            else:
+                # Windows will not generate SIGTERM itself
+                # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+                os.kill(old_pid, signal.SIGTERM)
+            return
+        elif opts.restart:
+            tools.stderr(
+                'options --restart is deprecated, '
+                'use `sopel restart` instead')
+            stderr('Asking Sopel to restart')
+            if hasattr(signal, 'SIGUSR2'):
+                os.kill(old_pid, signal.SIGUSR2)
+            else:
+                # Windows will not generate SIGILL itself
+                # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+                os.kill(old_pid, signal.SIGILL)
+            return
+    elif opts.kill or opts.quit or opts.restart:
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    if opts.daemonize:
+        child_pid = os.fork()
+        if child_pid is not 0:
+            return
+    with open(pid_file_path, 'w') as pid_file:
+        pid_file.write(str(os.getpid()))
+
+    # Step Five: Initialize and run Sopel
+    ret = run(config_module, pid_file_path)
+    os.unlink(pid_file_path)
+    if ret == -1:
+        os.execv(sys.executable, ['python'] + sys.argv)
+    else:
+        return ret
+
+
+def main(argv=None):
+    """Sopel run script entry point"""
+    try:
+        # Step One: Parse The Command Line
+        parser = build_parser()
+
+        # make sure to have an action first (`legacy` by default)
```

Comment:
Maybe tag this for updating? Not sure if necessary, but this ties in with my comment above about always doing _something useful_ when run with just `sopel` (no args), and `start` being a good default action.